akka-cluster/src/multi-jvm/scala/akka/cluster/RemoteFeaturesWithClusterSpec.scala

```diff
@@ -0,0 +1,125 @@
+  deployOn(third, """/kattdjur.remote = "@second@" """)
+
+}
+
+object ClusterRemoteFeatures {
+  class AddressPing extends Actor {
+    def receive: Receive = {
+      case "ping" => sender() ! self
+    }
+  }
+}
+
+class ArteryClusterRemoteFeaturesMultiJvmNode1 extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(true))
+class ArteryClusterRemoteFeaturesMultiJvmNode2 extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(true))
+class ArteryClusterRemoteFeaturesMultiJvmNode3 extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(true))
+
+class ClassicClusterRemoteFeaturesMultiJvmNode1
+    extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(false))
+class ClassicClusterRemoteFeaturesMultiJvmNode2
+    extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(false))
+class ClassicClusterRemoteFeaturesMultiJvmNode3
+    extends ClusterRemoteFeaturesSpec(new ClusterRemoteFeaturesConfig(false))
+
+abstract class ClusterRemoteFeaturesSpec(multiNodeConfig: ClusterRemoteFeaturesConfig)
+    extends MultiNodeSpec(multiNodeConfig)
+    with MultiNodeClusterSpec
+    with ImplicitSender
+    with ScalaFutures {
+
+  import multiNodeConfig._
+
+  override def initialParticipants: Int = roles.size
+
+  muteDeadLetters(Heartbeat.getClass)()
+
+  protected val provider: RemoteActorRefProvider = RARP(system).provider
+
+  "Remoting with Cluster" must {
+
+    "have the correct settings" in {
+      runOn(first) {
+        system.settings.HasCluster shouldBe true
+        provider.hasClusterOrUseUnsafe shouldBe true
+        provider.transport.system.settings.HasCluster shouldBe true
+        provider.remoteSettings.UseUnsafeRemoteFeaturesWithoutCluster shouldBe false
+        system.settings.ProviderClass shouldEqual classOf[ClusterActorRefProvider].getName
+      }
+    }
+
+    "create a ClusterRemoteWatcher" in {
+      runOn(roles: _*)(provider.remoteWatcher.isDefined shouldBe true)
+    }
+
+    "create a ClusterDeployer" in {
+      runOn(roles: _*)(provider.deployer.getClass shouldEqual classOf[ClusterDeployer])
+    }
+
+    "have expected `actorOf` behavior" in {
+      awaitClusterUp(first, second)
+      enterBarrier("cluster-up")
+
+      runOn(first) {
+        val actor = system.actorOf(Props[AddressPing], "kattdjur")
+        actor.isInstanceOf[RemoteActorRef] shouldBe true
+        actor.path.address shouldEqual node(second).address
+        actor.path.address.hasGlobalScope shouldBe true
+
+        val secondAddress = testConductor.getAddressFor(second).await
+        actor ! "ping"
+        expectMsgType[RemoteActorRef].path.address shouldEqual secondAddress
+      }
+      enterBarrier("CARP-in-cluster-remote-validated")
+
+      runOn(third) {
+        val actor = system.actorOf(Props[AddressPing], "kattdjur")
+        actor.isInstanceOf[RepointableActorRef] shouldBe true
+        val localAddress = AddressFromURIString(s"akka://${system.name}")
```

Comment:
ok, third has not joined cluster so it creates local, good

You can add one more test to verify that creating an actor from second with remote deployment to third it will also create a local actor. Add config:
```
deployOn(second, """/kattdjur.remote = "@third@" """)
```
and do that test from `runOn(second)`